libs/blocks/nav-marquee/nav-marquee.js

```diff
@@ -0,0 +1,129 @@
+    linkText.append(link);
+    item.addEventListener('click', () => {
+      const isTextSelected = window.getSelection().toString();
+
+      if (!isTextSelected) {
+        link.click();
+      }
+    });
+    link.addEventListener('click', (e) => {
+      const targetPosition = target?.getBoundingClientRect().top ?? 0;
+      const offsetPosition = targetPosition + window.pageYOffset - pageTop;
+
+      e.preventDefault();
+      window.scrollTo(0, offsetPosition);
+      target?.setAttribute('tabindex', -1);
+      target?.focus();
+    });
+  }
+
+  if (description) linkText.append(createTag('p', { class: 'section-description' }, description));
+  item.append(linkText);
+  item.append(createTag('div', { class: 'toc-arrow' }, DOWN_ARROW_ICON));
+  return item;
+}
+
+function decorateText(el, size) {
+  const headings = el.querySelectorAll('h1, h2, h3, h4, h5, h6');
+  const heading = headings[headings.length - 1];
+  const decorate = (headingEl, headingSize, bodySize, detailSize) => {
+    headingEl.classList.add(`heading-${headingSize}`);
+    headingEl.nextElementSibling?.classList.add(`body-${bodySize}`);
+    const sib = headingEl.previousElementSibling;
+    if (sib) {
+      sib.querySelector('img, .icon') ? sib.classList.add('icon-area') : sib.classList.add(`detail-${detailSize}`);
+      sib.previousElementSibling?.classList.add('icon-area');
+    }
+  };
+  size === 'large' ? decorate(heading, 'xxl', 'xl', 'l') : decorate(heading, 'xl', 'm', 'm');
+}
+
+export default function init(el) {
+  const size = getBlockSize(el);
+  const children = Array.from(el.querySelectorAll(':scope > div'));
+  const tone = (el.classList.contains('light')) ? 'light' : 'dark';
+  const marqueeContent = createTag('div', { class: 'foreground' }, children.shift());
+  const marquee = createTag('div', { class: `marquee ${tone}` }, marqueeContent);
+  marqueeContent.firstElementChild.classList.add('text');
+
+  decorateText(marquee, size);
+  decorateButtons(marquee, size === 'large' ? 'button-xl' : 'button-l');
+  marquee.className = `marquee ${tone}`;
+
+  const header = createTag('p', { class: 'toc-title' });
+  const footer = createTag('p', { class: 'toc-footer' });
+  const tocNav = createTag('nav', { 'aria-label': 'Table of contents' });
+  const navUl = createTag('ul', { class: 'toc-list' });
+  children.forEach((section) => {
+    if (section.firstElementChild.textContent === 'header') {
+      header.append(section.lastElementChild.textContent);
+      section.remove();
+      return;
+    }
+
+    if (section.firstElementChild.textContent === 'footer') {
+      footer.append(section.lastElementChild);
+      section.remove();
+      return;
+    }
+
+    const sectionTags = Array.from(section.querySelectorAll('p'));
+    const sectionTitle = section.querySelector('strong');
+    const link = section.querySelector('a');
+    const subtitle = sectionTags.find((element) => element.childElementCount === 0);
```

Comment:
```suggestion
    const subtitle = [...section.querySelectorAll('p')].find((element) => element.childElementCount === 0);
```
no need for the `sectionTags` const

Reply:
2 line reduction. great.